docs/source/tracking.rst

```diff
@@ -718,11 +718,14 @@ Azure Blob Storage
 To store artifacts in Azure Blob Storage, specify a URI of the form
 ``wasbs://<container>@<storage-account>.blob.core.windows.net/<path>``.
 MLflow expects Azure Storage access credentials in the
-``AZURE_STORAGE_CONNECTION_STRING`` or ``AZURE_STORAGE_ACCESS_KEY`` environment variables (preferring
-a connection string if one is set), so you must set one of these variables on both your client
-application and your MLflow tracking server. Finally, you must run ``pip install azure-storage-blob``
-separately (on both your client and the server) to access Azure Blob Storage; MLflow does not declare
-a dependency on this package by default.
+``AZURE_STORAGE_CONNECTION_STRING``, ``AZURE_STORAGE_ACCESS_KEY`` or all three of
+``AZURE_TENANT_ID`` & ``AZURE_CLIENT_ID`` & ``AZURE_CLIENT_SECRET`` environment variables
+(preferring a connection string over the access key, and the access key over the client credentials),
+so you must set one of these variables on both your client application and your MLflow tracking server.
+Finally, you must run ``pip install azure-storage-blob`` separately (on both your client and the server)
+to access Azure Blob Storage using the connection string or access key,
+or ``pip install azure-storage-blob azure-identity`` to access the storage via client credentials;
```

Comment:
(continued) ... this might be confusing to users if they didn't have to extend their installation already. Consider a more descriptive error on your usage by

```
try:
    from azure.identity import ....
except ImportError:
    <more descriptive error on package to install>
```

Reply:
Agreed. Added import to a try catch block and updated the documentation.